### PR TITLE
[BugFix] Fix orc writer write column that has no nulls

### PR DESCRIPTION
**Pull request by letian-jiang**
## What I'm doing:

Fix orc writer write column that has no nulls 

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/planner/HiveTableSink.java [hunk 1]
```diff
@@ -103,7 +104,11 @@ protected TDataSink toThrift() {
         tHiveTableSink.setStaging_dir(stagingDir);
         tHiveTableSink.setFile_format(fileFormat);
         tHiveTableSink.setIs_static_partition_sink(isStaticPartitionSink);
-        TCompressionType compression = PARQUET_COMPRESSION_TYPE_MAP.get(compressionType);
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/planner/HiveTableSink.java:106]: to pass correct compression type

File: be/src/formats/orc/orc_file_writer.cpp [hunk 2]
```diff
@@ -285,6 +285,9 @@ void ORCFileWriter::_write_number(orc::ColumnVectorBatch& orc_column, ColumnPtr&
         for (size_t i = 0; i < column_size; i++) {
             orc_column.notNull[i] = 1 - null_col[i];
         }
+    } else {
+        orc_column.hasNulls = false;
+        memset(number_orc_column.notNull.data(), 1, column_size * sizeof(char));
```
Comment (dirtysalt) [inline be/src/formats/orc/orc_file_writer.cpp:290]: number_orc_column or orc_column?

File: be/src/formats/orc/orc_file_writer.cpp [hunk 3]
```diff
@@ -599,6 +551,20 @@ StatusOr<std::unique_ptr<orc::Type>> ORCFileWriter::_make_schema_node(const Type
     }
 }
 
+void ORCFileWriter::_populate_orc_notnull(orc::ColumnVectorBatch& orc_column, uint8_t* null_column,
```
Comment (dirtysalt) [inline be/src/formats/orc/orc_file_writer.cpp:554]: can it be a static function? no need to be a method of class

Comment (letian-jiang) [reply be/src/formats/orc/orc_file_writer.cpp]: done

**Review by dirtysalt — DISMISSED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**